src/windows_utils.rs

```diff
@@ -1,222 +1,177 @@
+    s!("Software\\Classes\\Directory\\Background\\shell\\Neovide\\command");
+const REGISTRY_PATH_FOLDER: PCSTR = s!("Software\\Classes\\*\\shell\\Neovide");
+const REGISTRY_PATH_FOLDER_COMMAND: PCSTR = s!("Software\\Classes\\*\\shell\\Neovide\\command");
+
+fn get_neovide_path() -> String {
+    let mut buffer = vec![0u8; MAX_PATH as usize];
+    let len: u32;
     unsafe {
-        GetModuleFileNameA(
-            null_mut(),
-            buffer.as_mut_ptr() as *mut i8,
-            MAX_PATH as DWORD,
-        );
-        CString::from_vec_unchecked(buffer)
-            .into_string()
-            .unwrap_or_else(|_| "".to_string())
-            .trim_end_matches(char::from(0))
-            .to_string()
+        len = GetModuleFileNameA(HMODULE::default(), &mut buffer);
     }
+    buffer.truncate(len as usize);
+    String::from_utf8(buffer).unwrap()
 }
 
 fn unregister_rightclick() -> bool {
-    let str_registry_path_1 =
-        CString::new("Software\\Classes\\Directory\\Background\\shell\\Neovide").unwrap();
-    let str_registry_path_2 = CString::new("Software\\Classes\\*\\shell\\Neovide").unwrap();
     unsafe {
-        let s1 = RegDeleteTreeA(HKEY_CURRENT_USER, str_registry_path_1.as_ptr());
-        let s2 = RegDeleteTreeA(HKEY_CURRENT_USER, str_registry_path_2.as_ptr());
-        s1 == 0 && s2 == 0
+        RegDeleteTreeA(HKEY_CURRENT_USER, REGISTRY_PATH_DIRECTORY).0 == 0
+            && RegDeleteTreeA(HKEY_CURRENT_USER, REGISTRY_PATH_FOLDER).0 == 0
     }
 }
 
 fn register_rightclick_directory() -> bool {
-    let neovide_path = get_binary_path();
-    let mut registry_key: HKEY = null_mut();
-    let str_registry_path =
-        CString::new("Software\\Classes\\Directory\\Background\\shell\\Neovide").unwrap();
-    let str_registry_command_path =
-        CString::new("Software\\Classes\\Directory\\Background\\shell\\Neovide\\command").unwrap();
-    let str_icon = CString::new("Icon").unwrap();
-    let str_command = CString::new(format!("{} \"%V\"", neovide_path).as_bytes()).unwrap();
-    let str_description = CString::new("Open with Neovide").unwrap();
-    let str_neovide_path = CString::new(neovide_path.as_bytes()).unwrap();
+    let mut registry_key = HKEY::default();
+
+    let neovide_path = get_neovide_path();
+    let neovide_description = "Open with Neovide";
+    let neovide_command = format!("{} \"%V\"", neovide_path);
+
     unsafe {
         if RegCreateKeyExA(
             HKEY_CURRENT_USER,
-            str_registry_path.as_ptr(),
+            REGISTRY_PATH_DIRECTORY,
             0,
-            null_mut(),
+            PCSTR::null(),
             REG_OPTION_NON_VOLATILE,
-            KEY_WRITE,
-            null_mut(),
+            KEY_READ | KEY_WRITE,
+            None,
             &mut registry_key,
-            null_mut(),
-        ) != 0
+            None,
+        )
+        .0 != 0
         {
-            RegCloseKey(registry_key);
+            let _ = RegCloseKey(registry_key);
```

Comment:
The rust analyzer, recommends this pattern of `let _ = ` to ignore the result, but I think in our code base it's more common to use:
`RegCloseKey(registry_key).ok();`